Assets/IntegrationTests/TestShopifyCollections.cs

```diff
@@ -36,5 +37,41 @@ public class TestShopifyCollections : MonoBehaviour {
 
             Assert.IsTrue (waiter.IsStopped, Utils.MaxQueryMessage);
         }
+
+        [UnityTest]
+        public IEnumerator LoadAllCollections() {
+            float maxDuration = 6f;
+            int maxCollections = 13;
+            StoppableWaitForTime waiter = new StoppableWaitForTime (maxDuration);
+            ShopifyError errorsFromQueries = null;
+            List<Collection> AllCollections = new List<Collection>();
+
+            Action<string> loadPage = null;
+
+            loadPage = (string after) => {
+                Clients.GraphQLMany.collections((collections, errors, afterCollectionsLoaded) => {
+                    if (collections != null) {
+                        AllCollections.AddRange(collections);
+                    }
+
+                    if (errors != null) {
+                        errorsFromQueries = errors;
+                        waiter.Stop();
+                    } else if (afterCollectionsLoaded == null || AllCollections.Count >= maxCollections) {
+                        waiter.Stop();
+                    } else {
+                        loadPage(afterCollectionsLoaded);
+                    }
+                }, after: after);
+            };
```

Comment:
We could build this into the sdk at some point, the recursive paginator

Reply:
Yeah I was thinking about that to just return a lambda.